src/Symfony/Component/Dotenv/Dotenv.php

```diff
@@ -61,21 +61,28 @@ public function load($path, ...$paths)
     /**
      * Sets values as environment variables (via putenv, $_ENV, and $_SERVER).
      *
-     * Note that existing environment variables are never overridden.
-     *
      * @param array $values An array of env variables
      */
     public function populate($values)
     {
+        $loadedVars = getenv('SYMFONY_DOTENV_VARS') ? explode(',', getenv('SYMFONY_DOTENV_VARS')) : array();
+
         foreach ($values as $name => $value) {
-            if (isset($_ENV[$name]) || isset($_SERVER[$name]) || false !== getenv($name)) {
+            if (!in_array($name, $loadedVars) && (isset($_ENV[$name]) || isset($_SERVER[$name]) || false !== getenv($name))) {
                 continue;
             }
 
             putenv("$name=$value");
             $_ENV[$name] = $value;
             $_SERVER[$name] = $value;
+
+            $loadedVars[] = $name;
         }
+
+        $loadedVars = implode(',', $loadedVars);
```

Comment:
`if (1 < count($loadedVars)) { ... }` ("1" because of the empty string - or we should remove it before)